cmd/podman/common.go

```diff
@@ -39,24 +39,45 @@ func shortID(id string) string {
 	return id
 }
 
-// checkAllAndLatest checks that --all and --latest are used correctly
-func checkAllAndLatest(c *cobra.Command, args []string, ignoreArgLen bool) error {
+// checkAllAndLatest checks that --all and --latest are used correctly. If
+// cidfile is set, also check for the --cidfile flag.
+func checkAllAndLatest(c *cobra.Command, args []string, ignoreArgLen bool, cidfile bool) error {
```

Comment:
might be good to massage Cid into the function name.  perhaps checkCidAllAndLatest

Reply:
what about a specific function for it that wraps checkAllAndLatest?  That allows you to address Tom's comment and not change other usages? Only a suggestion.  Defer to your judgement,

Reply:
thinking about this more ... it might be a good idea because it would allow you to check for colliding flags, etc outside the checkAllAndLatest ... again, your call.

Reply:
I guess I'll go the lazy approach and follow `--storage` which also conflicts with `--all` and `--latest` but does the check outside the function. It's frustrating the `cobra` does not allow to specify conflicting flags. Maybe worth contributing?

Reply:
Eek, that doesn't work nicely either. Same applies to a wrapper func as we really need to check all flags together at once. It renamed the func though. Note that the additional parameter is needed for commands that don't support --cidfile ... it avoids printing errors that include --cidfile.